janusgraph-test/src/test/java/org/janusgraph/graphdb/query/QueryTest.java

```diff
@@ -127,5 +130,53 @@ public void testOrderList() {
     }
 
 
+    @Test
+    public void testMultipleIndexQueryWithLimits() {
+        JanusGraphManagement mgmt = graph.openManagement();
+        PropertyKey prop1Key = mgmt.makePropertyKey("prop1").dataType(String.class).make();
+        PropertyKey prop2Key = mgmt.makePropertyKey("prop2").dataType(String.class).make();
+
+        mgmt.buildIndex("prop1_idx", Vertex.class).addKey(prop1Key).buildCompositeIndex();
+        mgmt.buildIndex("prop2_idx", Vertex.class).addKey(prop2Key).buildCompositeIndex();
+
+        mgmt.commit();
+
+        //create 20 vertices with prop1=prop1val1, prop2=prop2val1
+        for(int i=0; i<20; i++)
+        {
+            tx.addVertex().property("prop1", "prop1val1").element().property("prop2", "prop2val1");
+        }
+        //and 1 vertex with prop1=prop1val1, prop2=prop2val2
+        tx.addVertex().property("prop1", "prop1val1").element().property("prop2", "prop2val2");
+
+        tx.commit();
+
+        List<Vertex> res;
+
+        //now query for that one vertex with prop1=prop1val1, prop2=prop2val2, with limit(1) - should still find it
+
+        //force to use no indexes - works ok
+        res = graph.traversal().V().map(x -> x.get()).has("prop2", "prop2val2").has("prop1", "prop1val1").limit(1).toList();
+        assertEquals(1, res.size());
+
+        //force to use only prop1 index - works ok
+        res = graph.traversal().V().has("prop1", "prop1val1").map(x -> x.get()).has("prop2", "prop2val2").limit(1).toList();
+        assertEquals(1, res.size());
+
+        //force to use only prop2 index - works ok
+        res = graph.traversal().V().has("prop2", "prop2val2").map(x -> x.get()).has("prop1", "prop1val1").limit(1).toList();
+        assertEquals(1, res.size());
+
+        //allow JanusGraphStep strategy to combine both has() into one graph step which tries to use both indexes
+        // without limit - works ok
+        res = graph.traversal().V().has("prop1", "prop1val1").has("prop2", "prop2val2").toList();
+        assertEquals(1, res.size());
+        // with limit - returns nothing before fix, works after fix
```

Comment:
Again very helpful comment for review, but it will be confusing out of context once the fix goes in.
```
// Tests with limit
```

Reply:
Thanks for the comments, I am happy to accept as is but unfortunately I don't seem to have correct permissions to accept your changes straight from web UI. will have to make them locally and update the PR tomorrow London time

Reply:
I'm not aware of a way to make review comments auto-mergable through the web UI, but that would be awesome. I plan to do more reviews tomorrow so I'll keep an eye out for the changes so I can approve. Thanks for the fix.